src/detect-engine-dns.c

```diff
@@ -112,19 +95,19 @@ static void PrefilterTxDnsQuery(DetectEngineThreadCtx *det_ctx,
 
     const MpmCtx *mpm_ctx = (MpmCtx *)pectx;
     DNSTransaction *tx = (DNSTransaction *)txv;
-    DNSQueryEntry *query = NULL;
 
-    TAILQ_FOREACH(query, &tx->query_list, next) {
-        SCLogDebug("tx %p query %p", tx, query);
+    for (uint16_t i = 0; i < 0xffff; i++) {
```

Comment:
I like this approach better as for me it keeps a cleaner boundary between rust and C. Don't really like this loop 'i<64k' but I guess that isn't even needed as we'll never get that many records. We can probably just make it an endless loop as we'll break out because of rs_dns_tx_get_query_buffer retval.